concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,214 @@
+    cluster_url: str,
+    report_computed_task: ReportComputedTask,
+    expected_status: Optional[int] = None,
+) -> None:
+    response = api_request(
+        cluster_url,
+        'send',
+        REQUESTOR_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_force_get_task_result(
+            report_computed_task=report_computed_task,
+        ),
+        headers={
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=expected_status,
+    )
+    responses_global.append(response.__class__.__name__)
+
+
+@count_fails
+def test_case_multiple_requests_concerning_one_subtask_will_be_processed_one_by_one_if_subtask_exists_in_database(
+    cluster_consts: str,
+    cluster_url: str,
+    task_id: str,
+    subtask_id: str
+):
+    report_computed_task = create_report_computed_task(task_id=task_id, subtask_id=subtask_id)
+
+    """send messages to save subtask in database and change state to REPORTED. Next message ForceGetTaskResult
+       requires REPORTED state and this is the simplest way to get it """
+    send_correct_force_report_computed_task(
+        cluster_url=cluster_url,
+        report_computed_task=report_computed_task,
+        expected_status=202,
+    )  # Subtask state changed to: FORCING_REPORT
+    send_correct_ack_report_computed_task(
+        cluster_url=cluster_url,
+        report_computed_task=report_computed_task,
+        expected_status=202,
+    )  # Subtask state changed to: REPORTED
+
+    clear_responses()
+    """this is test- sending some messages in one time """
+    call_function_in_treads(
+        func=send_correct_force_get_task_result,  # Subtask state changed to: FORCING_RESULT_TRANSFER
+        number_of_threads=NUMBER_OF_TESTING_THREADS,
+        cluster_url=cluster_url,
+        golem_message=report_computed_task,
+    )
+    "waiting for responses from all threads"
+    end_time = time.time() + MAXIMUM_WAITING_TIME_FOR_ALL_RESPONSES
+    while len(responses_global) != NUMBER_OF_TESTING_THREADS:
+        time.sleep(0.1)
+        if time.time() >= end_time:
+            break
+
+    expected_responses = ['AckForceGetTaskResult', 'ServiceRefused', 'ServiceRefused']
+
+    print('Responses = ' + str(responses_global))
+    assert_condition(
+        actual=responses_global,
+        expected=expected_responses,
+        error_message=f"Responses should be: {expected_responses}. Probably reason is that something is wrong with code"
+        f" and test failed or maybe you changed number of testing threads and forgot to change expected_responses?",
+        compare_lists_regardless_of_order=True,
+    )
+    print('Single test passed successfully')
+
+
+if __name__ == '__main__':
+    try:
+        from concent_api.settings import CONCENT_PUBLIC_KEY
+
+        run_tests(globals())
+    except requests.exceptions.ConnectionError as exception:
+        print("\nERROR: Failed connect to the server.\n", file=sys.stderr)
+        sys.exit(str(exception))
```

Comment:
I think that `exit(exception)` would work too.

Reply:
done